tests/test_edgeql_casts.py

```diff
@@ -2162,6 +2162,32 @@ async def test_edgeql_casts_json_10(self):
             await self.con.query_one(
                 'SELECT <cal::local_time><json>"20:01:22.306916+01:00"')
 
+    async def test_edgeql_casts_json_11(self):
+        await self.assert_query_result(
+            r"SELECT <array<int64>><json>[1, 1, 2, 3, 5]",
+            [[1, 1, 2, 3, 5]]
+        )
+
+        with self.assertRaisesRegex(
+                # FIXME: This should be a different error
+                edgedb.InternalServerError,
```

Comment:
The simplest way would be to map these via `errormech.py` to `errors.InvalidValueError`

Reply:
OK. I'll do a follow-up PR that improves a bunch of JSON related errors.

Reply:
Yes, we need to map `wrong_object_type` (42809) to `InvalidValueError`.  That can be done in a separate PR, though.